docs/core/reference/versioning.md

```diff
@@ -60,7 +60,17 @@ Software packages are mutually compatible if they share the same major and minor
 
 #### Deprecations and breaking changes
 
-Deprecations and breaking changes can occur when the minor version changes. If a feature is deprecated, it will remain functional for at least one minor version before being removed.
+Deprecations and breaking changes can occur when the minor version changes. To the greatest extent possible, breaking changes in server packages will be preceded by a deprecation announcement and at least one minor version cycle of continued support.
```

Comment:
I would remove "to the greatest extent possible". We should keep a strong promise for client-server compatibility, not hedge our language to give us wiggle room in the future.